src/pybind/mgr/orchestrator/_interface.py

```diff
@@ -987,7 +987,7 @@ def upgrade_ls(self, image: Optional[str], tags: bool, show_all_versions: Option
         raise NotImplementedError()
 
     def upgrade_start(self, image: Optional[str], version: Optional[str], daemon_types: Optional[List[str]],
-                      hosts: Optional[str], services: Optional[List[str]], limit: Optional[int]) -> OrchResult[str]:
+                      hosts: Optional[str], services: Optional[List[str]], limit: Optional[int], set_mon_flags: bool = False) -> OrchResult[str]:
```

Comment:
This seems to be the only place we're using the positive "set mon flags" and then everywhere else it's the negative "no mon flags". Would it be possible to keep using the positive statement through the full path from here to the actual upgrade code?

Reply:
Using `unset_mon_flags` throughout so there is no discrepancy.